scripts/openmc-make-depletion-chain-casl

```diff
@@ -0,0 +1,240 @@
+    fpy_data = {}
+    for f in fpy_files:
+        data = openmc.data.FissionProductYields(f)
+        name = data.nuclide['name']
+        if name in CASL_CHAIN:
+            fpy_data[name] = data
+
+    print('Creating depletion_chain...')
+    missing_daughter = []
+    missing_rx_product = []
+    missing_fpy = []
+
+    for idx, parent in enumerate(sorted(decay_data, key=openmc.data.zam)):
+        data = decay_data[parent]
+
+        nuclide = Nuclide()
+        nuclide.name = parent
+
+        chain.nuclides.append(nuclide)
+        chain.nuclide_dict[parent] = idx
+
+        if not CASL_CHAIN[parent][0] and \
+           not data.nuclide['stable'] and data.half_life.nominal_value != 0.0:
+            nuclide.half_life = data.half_life.nominal_value
+            nuclide.decay_energy = sum(E.nominal_value for E in
+                                       data.average_energies.values())
+            sum_br = 0.0
+            for i, mode in enumerate(data.modes):
+                type_ = ','.join(mode.modes)
+                if mode.daughter in decay_data:
+                    target = mode.daughter
+                else:
+                    print('missing {} {} {}'.format(parent, ','.join(mode.modes), mode.daughter))
+                    continue
+
+                # Write branching ratio, taking care to ensure sum is unity
+                br = mode.branching_ratio.nominal_value
+                sum_br += br
+                if i == len(data.modes) - 1 and sum_br != 1.0:
+                    br = 1.0 - sum(m.branching_ratio.nominal_value
+                                   for m in data.modes[:-1])
+
+                # Append decay mode
+                nuclide.decay_modes.append(DecayTuple(type_, target, br))
+
+        if parent in reactions:
+            reactions_available = set(reactions[parent].keys())
+            for name, mts, changes in _REACTIONS:
+                if mts & reactions_available:
+                    delta_A, delta_Z = changes
+                    A = data.nuclide['mass_number'] + delta_A
+                    Z = data.nuclide['atomic_number'] + delta_Z
+                    daughter = '{}{}'.format(openmc.data.ATOMIC_SYMBOL[Z], A)
+
+                    if name not in chain.reactions:
+                        chain.reactions.append(name)
+
+                    if daughter not in decay_data:
+                        missing_rx_product.append((parent, name, daughter))
+                        daughter = 'Nothing'
+
+                    # Store Q value
+                    for mt in sorted(mts):
+                        if mt in reactions[parent]:
+                            q_value = reactions[parent][mt]
+                            break
+                    else:
+                        q_value = 0.0
+
+                    nuclide.reactions.append(ReactionTuple(
+                        name, daughter, q_value, 1.0))
+
+            if any(mt in reactions_available for mt in [18, 19, 20, 21, 38]):
```

Comment:
Optional: I would leave a comment clarifying the significance of the `mt` values `[18, 19, 20, 21, 38]` (by the logic, they look related to missing fission product yields?).  Or create a file-level constant with this list.